src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -146,6 +146,32 @@ internal static MsalTokenResponse CreateFromiOSBrokerResponse(Dictionary<string,
             return response;
         }
 
+        internal static MsalTokenResponse CreateFromAppProviderResponse(ExternalTokenResult tokenResponse)
+        {
+            var response = new MsalTokenResponse
+            {
+                AccessToken = tokenResponse.RawAccessToken,
+                RefreshToken = null,
+                IdToken = null,
+                TokenType = BrokerResponseConst.Bearer,
+                CorrelationId = tokenResponse.CorrelationId,
```

Comment:
I don't see the point in getting a new correlation id from the "idp". I think in MSAL we just rely on the fact that the ESTS gives us back the same correlation id.

Reply:
The method above uses the same logic as it gets the correlation id from the response on line 128.
CorrelationId = responseDictionary[BrokerResponseConst.CorrelationId],

Technically this line should also simply inject the correlation id from the input params then.